src/platforms/windows/windowseventlistener.cpp

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "windowseventlistener.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "platforms/windows/windowscommons.h"
+#include "qmlengineholder.h"
+
+#include <QLocalSocket>
+
+#include <Windows.h>
+
+constexpr const char* UI_PIPE = "\\\\.\\pipe\\mozillavpn.ui";
+
+namespace {
+Logger logger(LOG_WINDOWS, "WindowsEventListener");
+}
+
+WindowsEventListener::WindowsEventListener() {
+  logger.log() << "Windows event listener created";
+
+  m_server.setSocketOptions(QLocalServer::WorldAccessOption);
```

Comment:
Do we need WorldAccessOption? Would this make the pipe exposed via smb to the local network?

Reply:
Good catch! I rebased this patch on top of main. Closing and reopening